R/zchunk_LA123.electricity.R

```diff
@@ -29,26 +29,118 @@ module_energy_LA123.electricity_DISABLED <- function(command, ...) {
+      select(fuel,electricity)%>%
+      filter(!is.na(electricity))->enduse_fuel_aggregation_electricity
+
+    # Creates L123.in_EJ_R_elec_F_Yh based on L1011.en_bal_EJ_R_Si_Fi_Yh and enduse_fuel_aggregation_electricity
+    # Calculates the inpunts by fuel (based on electricity input fuels), region ID and sector (electricity)
+    L1011.en_bal_EJ_R_Si_Fi_Yh%>%
+      filter(sector == "in_electricity generation")%>%
+      mutate(sector = "electricity generation")%>%
+      left_join(enduse_fuel_aggregation_electricity,by="fuel")%>%
+      mutate(fuel=electricity)%>%
+      select(-electricity)%>%
+      filter(fuel %in% ELECTRICITY_INPUT_FUELS)%>%
+      group_by(GCAM_region_ID, sector,fuel)%>%
+      summarise_each(funs(sum))->L123.in_EJ_R_elec_F_Yh
+
+    # Creates L123.out_EJ_R_elec_F_Yh based on L1011.en_bal_EJ_R_Si_Fi_Yh and enduse_fuel_aggregation_electricity
+    # Calculates the electricity outputs by fuel, region ID and sector (electricity generation)
+    L1011.en_bal_EJ_R_Si_Fi_Yh%>%
+      filter(sector == "out_electricity generation")%>%
+      mutate(sector = "electricity generation")%>%
+      left_join(enduse_fuel_aggregation_electricity,by="fuel")%>%
+      mutate(fuel=electricity)%>%
+      select(-electricity)%>%
+      group_by(GCAM_region_ID, sector,fuel)%>%
+      summarise_each(funs(sum))%>%
+      filter(!is.na(fuel))->L123.out_EJ_R_elec_F_Yh
+
+
+    # Long form of L123.in_EJ_R_elec_F_Yh (previously created) to be used to calculate efficiencies
+    # based on electricity inputs and outputs by fuel and year.
+    L123.in_EJ_R_elec_F_Yh%>%
+      gather(X_HISTORICAL_YEARS,input,-GCAM_region_ID,-sector,-fuel)->input_fuels_by_year_region
+
+    # Calculates electricity generation efficiencies (L123.eff_R_elec_F_Yh) by fuel and year
+    L123.out_EJ_R_elec_F_Yh%>%
+      semi_join(L123.in_EJ_R_elec_F_Yh,by=c("GCAM_region_ID", "fuel"))%>%
+      gather(X_HISTORICAL_YEARS,outputs,-GCAM_region_ID,-sector,-fuel)%>%
+      left_join(input_fuels_by_year_region, by=c("GCAM_region_ID","sector","fuel","X_HISTORICAL_YEARS"))%>%
+      mutate(eff=outputs/input)%>%
+      select(-outputs,-input)%>%
+      spread(X_HISTORICAL_YEARS,eff)->L123.eff_R_elec_F_Yh
+
+    # Taking care of NA, 0, and INF values generatted in previous step (efficiency calculations) and updating
+    # the efficiency output L123.eff_R_elec_F_Yh
+    L123.eff_R_elec_F_Yh%>%
+      mutate_at(funs(replace(., is.na(.), DEFAULT_ELECTRIC_EFFICIENCY)), .cols = X_HISTORICAL_YEARS)%>%
+      mutate_at(funs(replace(., .==0, DEFAULT_ELECTRIC_EFFICIENCY)), .cols = X_HISTORICAL_YEARS)%>%
+      mutate_at(funs(replace(., .==Inf, DEFAULT_ELECTRIC_EFFICIENCY)), .cols = X_HISTORICAL_YEARS)->L123.eff_R_elec_F_Yh
+
+    # Tibble (Output_efficiency_based) created to adjust electricity outputs (L123.out_EJ_R_elec_F_Yh)
+    # based on above modified efficiency calculations
+    L123.in_EJ_R_elec_F_Yh%>%
+      gather(X_HISTORICAL_YEARS,outputs,-GCAM_region_ID,-sector,-fuel)%>%
```

Comment:
I "gather" all tibbles and used them for calculations. I "spread" them before creating the output files and saving them.